src/TimeGrid.jsx

```diff
@@ -93,7 +93,11 @@ let TimeGrid = React.createClass({
 
     allDayEvents.sort((a, b) => sortEvents(a, b, this.props))
 
-    let segments = allDayEvents.map(evt => eventSegments(evt, start, end, this.props))
+    var _endOfRange = endOfRange(range);
+    var first = _endOfRange.first;
+    var last = _endOfRange.last;
+
+    let segments = allDayEvents.map(evt => eventSegments(evt, first, last, this.props))
```

Comment:
thanks! would you mind writing this as destructuring?

`let { first, last } = endOfRange(range);`

its a bit less noisy, not a big deal if you can't